helper/fairshare/jobmanager.go

```diff
@@ -292,6 +287,10 @@ func (j *JobManager) assignWork() {
 				default:
 				}
 
+				if j.paused.Load() {
```

Comment:
it looks like this would just break to the outer for loop, and then we'd have a really tight loop burning CPU (unless I'm missing something)

this could be avoided by having some sort of sleep before the break (so that it doesn't affect the outer loop in the case where we aren't paused)